README.md

```diff
@@ -83,7 +83,7 @@ cargo run --example mnist --release
 fn main() -> HashMap<String, Vec<f32>> {
     let mut input_data = HashMap::new();
     let image = load_squeezenet_image(); // Load image
-    input_data.insert("data".to_string(), image.as_slice().unwrap());
+    input_data.insert("data".to_string(), InputTensor::F32(image.as_slice().unwrap()));
```

Comment:
Shall we not be able to do inference on the data type by using `From` `Into`, removing the need for the user to specify again the type of the tensor.

Reply:
That would be nicer indeed (it would look like `image.as_slice().unwrap().into()`).

Reply:
Well I was even thinking about putting the into within the function with the input hasmap value having a signature of type generic `impl Into`. Not sure what the implications are, but seems more logical for me.

Reply:
> If you want to feed a model inputs of different type (e.g. one F32 tensor and one I32 tensor, which happens sometimes) then you will be back to InputTensor anyway as it allows you to pass either in the same HashMap.

That is true.

Ok, I tried generics as well and it was indeed not the best solution.

The builder pattern looks interesting. 

I'm ok to keep the API at the current then.

Good discussion